service/src/main/java/bio/terra/catalog/common/StorageSystemService.java

```diff
@@ -0,0 +1,72 @@
+package bio.terra.catalog.common;
+
+import bio.terra.catalog.model.DatasetPreviewTable;
+import bio.terra.catalog.model.SystemStatusSystems;
+import bio.terra.catalog.model.TableMetadata;
+import bio.terra.catalog.service.dataset.DatasetAccessLevel;
+import bio.terra.common.iam.AuthenticatedUserRequest;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * An implementation of a storage system for catalog. A storage system adds support to catalog by
+ * implementing these APIs.
+ */
+public interface StorageSystemService {
+
+  /**
+   * @return an object indicating the system's status
+   */
+  SystemStatusSystems status();
+
+  /**
+   * Return all objects in the storage system that are visible to the user, and the storage system
+   * information for each object.
+   *
+   * @param user the user
+   * @return a map of storage object ID to access level
+   */
+  Map<String, StorageSystemInformation> getObjects(AuthenticatedUserRequest user);
```

Comment:
I _think_ there's a more descriptive name here than `getObjects`. Maybe `getStorageSystemData`?

Reply:
I'd rather not repeat words that are already in the class name. So I don't think it should contain `storage` or `system`. Would `getData()` be better?  The other suggestion I had was `getDatasets()`, since it's returning the objects from the storage system that are datasets to the catalog.